camelot/core/files/storage.py

```diff
@@ -276,27 +280,28 @@ def delete(self, name: PurePosixPath):
         Path(self.path(name)).unlink(missing_ok=True)
 
 
+def get_hashed_path(path: PurePosixPath) -> str:
+    return sha1(path.name.encode('UTF-8')).hexdigest()
+
+
 class HashStorage(Storage):
 
-    def __init__(self, upload_to: PurePosixPath = None):
+    def __init__(self, upload_to: PurePosixPath):
         super().__init__(upload_to)
-        self.hash_folder: PurePosixPath = PurePosixPath('')
 
-    def _create_tempfile(self, suffix: str, prefix: str):
-        # @todo suffix and prefix should be cleaned, because the user might be
-        #       able to get directory separators in here or something related
-        """Create a temporary file in the storage directory
+    def _create_tempfile(self, suffix: str, prefix: str, dir_: str = None) -> (int, str):
+        hashed_prefix = get_hashed_path(PurePosixPath(prefix))
+        return tempfile.mkstemp(suffix=suffix, prefix=hashed_prefix, dir=self.upload_to/hashed_prefix[:2])
+
+    def checkin(self, local_path: PurePosixPath, filename: PurePosixPath = None) -> StoredFile:
+        file = super().checkin(local_path, filename)
+        hashed_prefix = get_hashed_path(file.name)
+        return StoredFile(self, PurePosixPath(hashed_prefix[:2], hashed_prefix))
```

Comment:
voor de HashStorage moet de 'upload_to' component ook mee in de db komen te zitten, dwz dat die OF : 

- hier mee in de name vd StoredFile moet zitten
- er moet worden bijgeplakt in de File serialisatie

ik veronderstel dat het meest eenvoudige gaat zijn om die er hier bij te plakken

Reply:
Wat is het nut van heel het upload_to component in de db te steken? we weten toch waar de base folder zit? of is dit voor het geval dat de base folder zou veranderen